test/parallel/test-vm-module-synthetic.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+// Flags: --experimental-vm-modules
+
+require('../common');
+const { SyntheticModule, SourceTextModule } = require('vm');
+const assert = require('assert');
+
+(async () => {
+  const s = new SyntheticModule(['x'], () => {
+    s.setExport('x', 1);
```

Comment:
We need to ensure we have test cases here for:

1. Errors thrown during the synthetic execution
2. `setExport` being called before linking
3. `setExport` being called for an export that is not a valid export name.

Ideally none of the above should segfault and should have vaguely usable errors!

Reply:
will do. currently V8 aborts if the name is invalid but it should be throwing a ReferenceError. I posted a comment on the tracking issue so hopefully that will be solved soon.